sky/serve/replica_managers.py

```diff
@@ -46,12 +46,13 @@
 
 # Since sky.launch is very resource demanding, we limit the number of
 # concurrent sky.launch process to avoid overloading the machine.
-_MAX_NUM_LAUNCH = psutil.cpu_count()
+_MAX_NUM_LAUNCH = psutil.cpu_count() * 2
```

Comment:
The original max launch of `cpu_count()` seems a bit too conserverative, I increased it to 2x, and seems `pytest tests/test_smoke.py --serve` works fine when multiple serving tests are running in parallel.